src/components/managerlib/pathutils.ts

```diff
@@ -8,9 +8,11 @@ import type {Extension} from '../../main'
 
 export class PathUtils {
     private readonly extension: Extension
+    readonly inputRegex: RegExp
 
     constructor(extension: Extension) {
         this.extension = extension
+        this.inputRegex = /(?:(?:\\(?:input|InputIfFileExists|include|SweaveInput|subfile|(?:(?:sub)?(?:import|inputfrom|includefrom)\*?{([^}]*)}))(?:\[[^[\]{}]*\])?){([^}]*)})|(?:<<(?:[^,]*,)*\s*child='([^']*)'\s*(?:,[^,]*)*>>=)/g
```

Comment:
Making a regexp longer increases the maintenance burden. We should make it shorter if possible.

For example, with the following prototype, we can keep each regexp, `regA` and `regB`, short.
```ts
class A {
    private regA: RegExp
    private regB: RegExp

    match(s: string) {
        let match = this.regA.exec(s)
        if (match) {
            ...
            return ...
        } 
        match = this.regB.exec(s)
        if (match) {
            ...
            return ...
        }
    }
}
```

Reply:
I do agree. Indeed, the real maintenance burden is 

https://github.com/James-Yu/LaTeX-Workshop/blob/bfd87dadd0d01d1ff4a003b0e2257a7c45d67899/src/providers/structure.ts#L94-L101

especially because of the double escaping. It would be better to split the regex into a sectioning regex and an input regex. I will think of it